services/asymkey/ssh_key.go

```diff
@@ -27,20 +27,9 @@ func DeletePublicKey(ctx context.Context, doer *user_model.User, id int64) (err
 		}
 	}
 
-	dbCtx, committer, err := db.TxContext(ctx)
```

Comment:
Or here?

Reply:
This operation only involves a single DELETE statement, so wrapping it in a transaction is unnecessary. The database guarantees atomicity for individual SQL statements by default.